edgexfoundry/app-record-replay
# feat: Implement app layer Start Recording

closes #17







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/app-record-replay/blob/main/.github/CONTRIBUTING.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I have opened a PR for the related docs change (if not, why?) **TBD**
  <link to docs PR>

## Testing Instructions
Run non-secure EdgeX stack with Device Virtual
build  run app service 
run `WRITABLE_LOGLEVEL=DEBUG ./app-record-replay -cp -d | grep -e "ARR Start Recording" -e "ARR Process Recorded Data"` 
Post following to `localhost:59712/api/v3/record`
```
{
    "duration" : 0,
    "eventLimit" : 10
}
```
Using debug log messages verify recording ends after 10 events recorded 
Post following to `localhost:59712/api/v3/record`
```
{
    "duration" : 6000000000,
    "eventLimit" : 0
}
```
Using debug log messages verify recording ends after X events recorded in about 6s 
Post the following twice quickly to `localhost:59712/api/v3/record`
```
{
    "duration" : 6000000000,
    "eventLimit" : 0
}
```
Verify second request result in `5005` error with following message
```
Recording failed: a recording is already in progress
```
## New Dependency Instructions (If applicable)
